common/djangoapps/course_modes/models.py

```diff
@@ -776,7 +776,7 @@ def is_eligible_for_certificate(cls, mode_slug, status=None):
         """
         ineligible_modes = [cls.AUDIT]
 
-        if settings.FEATURES['DISABLE_HONOR_CERTIFICATES']:
+        if settings.FEATURES.get('DISABLE_HONOR_CERTIFICATES'):
```

Comment:
For my personal knowledge, is there a reason this was updated to use `get` over accessing directly by key?

Is it avoid the possibility of an exception being thrown? (It seems with `get` if the key doesn't exist the value would be None, where accessing directly can throw a `KeyError` exception).

Just curious if there was something else I may be missing :)

Reply:
I want to avoid a Key error here when the setting does not exist in the file. Get will return None by default and this will be False, which will have the same effect. The KeyError was not providing value.

With this check as is if the setting is not present the exception causes a failure to trickle through to the dashboard.